webapp/lib/main_ui.dart

```diff
@@ -45,9 +53,88 @@ class CodaUI {
     dataset.messages.forEach((message) {
       MessageViewModel messageViewModel = new MessageViewModel(message, dataset);
       messages.add(messageViewModel);
+      messageMap[message.id] = messageViewModel;
       body.append(messageViewModel.viewElement);
     });
     messageCodingTable.append(body);
+
+    messageCodingTable.onChange.listen((event) {
+      var target = event.target;
+      if (target is! InputElement && target is! SelectElement) return;
+
+      TableRowElement row = getAncestors(CodeSelector.activeCodeSelector.viewElement).firstWhere((e) => e.classes.contains('message-row'));
+      DivElement inputGroup = getAncestors(CodeSelector.activeCodeSelector.viewElement).firstWhere((e) => e.classes.contains('input-group'));
+      String messageID = row.attributes['message-id'];
+      MessageViewModel message = messageMap[messageID];
+      String schemeID = inputGroup.attributes['scheme-id'];
+
+      if (target is InputElement) { // change on checkbox element
+        message.schemeCheckChanged(dataset, schemeID, target.checked);
```

Comment:
I think you can add a shortcut return here, or turn the if statements into a switch

Reply:
I'll add a shortcut return, switching would result in losing the type of the target element useful for not needing to do a cast for target.checked.